### PR TITLE
Upgrade Gradle plugin and PyCharm to latest

### PR DESCRIPTION
**Pull request by intgr**
### Contributor checklist

- [x] I am targeting the `master` branch (and **not** the `release` branch)
- [x] I am using the provided [codeStyleConfig.xml](https://github.com/leinardi/mypy-pycharm/blob/release/.idea/codeStyles/codeStyleConfig.xml)
- [x] I have tested my contribution on these versions of PyCharm/IDEA:
 * PyCharm Community 2021.3
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit 
      using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-using-keywords/)

----------

### Description
* Upgraded to PyCharm 2021.3
* Upgrade IntelliJ Gradle plugin to 1.4.0.
* Changed `build.gradle` to work with latest plugin: https://lp.jetbrains.com/gradle-intellij-plugin/#features-392

### Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :hammer: Refactoring  |

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -67,32 +67,32 @@ sourceCompatibility = JavaVersion.VERSION_1_8
...
-    version ideVersion
-    pluginName props.getProperty('plugin.name').toLowerCase().replace(' ', '-')
-    downloadSources Boolean.valueOf(downloadIdeaSources)
+    version = ideVersion
+    pluginName = props.getProperty('plugin.name').toLowerCase().replace(' ', '-')
+    downloadSources = Boolean.valueOf(downloadIdeaSources)
     updateSinceUntilBuild = true
     if (hasPycharmPath) {
-        alternativeIdePath pycharmPath
+        alternativeIdePath.set(pycharmPath)
```
Comment (leinardi) [inline build.gradle:75]: Hi @intgr, just curious: why using `.set()` here instead of equal like in the other lines?

**Comment by leinardi** [comment]
LGTM, just curious about the `.set()`.

File: build.gradle (hunk 1 above)
Comment (intgr) [reply build.gradle:75]: Ah this is just an oversight, I was initially confused about the differences between Groovy and Kotlin DSL.

**Review by leinardi — APPROVED** [review]
Thanks!

**Pull request merged**